packages/qwik/src/core/qrl/qrl.ts

```diff
@@ -225,26 +238,16 @@ export const serializeQRLs = (
   return mapJoin(existingQRLs, (qrl) => serializeQRL(qrl, opts), '\n');
 };
 
-/** `./chunk#symbol[captures] */
+/** `./chunk#[attr#][symbol][[captures]] */
 export const parseQRL = <T = any>(qrl: string, containerEl?: Element): QRLInternal<T> => {
-  const endIdx = qrl.length;
-  const hashIdx = indexOf(qrl, 0, '#');
-  const captureIdx = indexOf(qrl, hashIdx, '[');
-
-  const chunkEndIdx = Math.min(hashIdx, captureIdx);
-  const chunk = qrl.substring(0, chunkEndIdx);
-
-  const symbolStartIdx = hashIdx == endIdx ? hashIdx : hashIdx + 1;
-  const symbolEndIdx = captureIdx;
-  const symbol =
-    symbolStartIdx == symbolEndIdx ? 'default' : qrl.substring(symbolStartIdx, symbolEndIdx);
-
-  const captureStartIdx = captureIdx;
-  const captureEndIdx = endIdx;
-  const capture =
-    captureStartIdx === captureEndIdx
-      ? EMPTY_ARRAY
-      : qrl.substring(captureStartIdx + 1, captureEndIdx - 1).split(' ');
+  const parse = /^(?<c>[^#[]*)#?((?<a>[^#]+)#)?(?<s>[^[]*)(\[(?<p>[^\]]*)\])?$/.exec(qrl);
```

Comment:
compiling regexp takes time so maybe this regexp could be outside the function?

Reply:
no longer relevant, but it turns out that inline regex is faster than regex in const! https://www.measurethat.net/Benchmarks/Show/31074/0/regex-defined-inside-vs-outside-loop